compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmConstants.java

```diff
@@ -382,7 +382,7 @@ public class JvmConstants {
     public static final String LAMBDA_PREFIX = "$lambda$";
     public static final String POPULATE_METHOD_PREFIX = "$populate";
     public static final String ADD_METHOD = "add";
-    public static final String TEST_EXECUTION_STATE = "$testExecutionState";
+    public static final String TEST_EXECUTION_STATE = "__testExecState__";
```

Comment:
We need to revert this to `"$testExecutionState"` otherwise we get a compilation error with the line `int __testExecState__ = 0;` in the test file. See
```
Compiling source
        nadeeshand/baltestissue:0.1.0
ERROR [tests/test_execute-generated_1.bal:(2:12,2:29)] redeclared symbol '__testExecState__'
error: compilation contains errors
```

Reply:
Since we are actually generating a bal code we can't have a variable name with `$` there. Even we can't use `__execute__` as a function name in our bal files as well since a function with that name has been created to call the test suite.

Reply:
Since we are generating Ballerina code and we need to have a valid Ballerina identifier here, we will keep this as `"__testExecState__"`. We will add this as a breaking change in the release note. (i.e. it will now give compilation errors if someone declare an identifier with name `__testExecState__`.)

Reply:
For safety and to reduce the probability of collisions with user's variable naming shall we also add some random characters to this also (something like "__gH7W16ntestExecState__", we can add a separate the part "gH7W16n" to a constant and use it here for identifying easily) and use the same TEST_EXECUTION_STATE constant in the 2 places in the compiler plugin code also where it is used.